apollo-api/src/jvmMain/kotlin/com/apollographql/apollo3/api/FileUpload.kt

```diff
@@ -5,8 +5,7 @@ import okio.buffer
 import okio.source
 import java.io.File
 
-fun Upload.Companion.fromFile(file: File, contentType: String) = object : Upload {
-  override val contentType = contentType
+class FileUpload(private val file: File, override val contentType: String) : Upload {
```

Comment:
However the `fromFile` naming was consistent with the other `fromXyz` methods in the `Upload` interface.

Reply:
True. I was thinking of changing them to things like `ByteStringUpload` like for HttpBody https://github.com/apollographql/apollo-android/blob/f5eb27ade6185544bfa2958bfda6671172ca3b66/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/http/ByteStringHttpBody.kt#L7

I don't think we have a lot of `fromXyz` and TBH I don't like them too much because they:

1 create a asymmetry between the builtin implementation vs others that can be defined in other modules
2 break the typing flow

Having `ByteStringUpload` also breaks the typing flow but at least it makes it easy to see all implementations of a given interface. 

If we wanted things to be even more fluent, we could then add things like `fun String.toUpload(): Upload`. This is what we do in `apollo-ast` but I'm hesitant doing this on super common types like `String`

Reply:
Yes, making the same change on the other ones is 👍 

(ps: you probably didn't forget, but just in case, reminder to impact the Upload doc 😊)

Reply:
Oh yes, I 100% forgot already! Thanks for the reminder!